apps/explorer/src/pages/validator/ValidatorDetails.tsx

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { is, SuiObject, Base64DataBuffer } from '@mysten/sui.js';
+import { useMemo } from 'react';
+import { useParams, Navigate } from 'react-router-dom';
+
+import ErrorResult from '~/components/error-result/ErrorResult';
+import { StatsCoin } from '~/components/top-validators-card/StatsCoin';
+import { useGetObject } from '~/hooks/useGetObject';
+import {
+    VALIDATORS_OBJECT_ID,
+    type ValidatorState,
+} from '~/pages/validator/ValidatorDataTypes';
+import { Card } from '~/ui/Card';
+import { DescriptionList, DescriptionItem } from '~/ui/DescriptionList';
+import { Heading } from '~/ui/Heading';
+import { ImageIcon } from '~/ui/ImageIcon';
+import { AddressLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Stats } from '~/ui/Stats';
+import { Text } from '~/ui/Text';
+import { getName } from '~/utils/getName';
+import { getStakedPercent } from '~/utils/getStakedPercent';
+
+function ValidatorDetails() {
```

Comment:
I think this is a good candidate to break into some smaller components (e.g. stats). They don't even need to be re-usable UI components, just some components to keep this cleaner / encourage more re-use.